src/views/AddItem.jsx

```diff
@@ -1,3 +1,87 @@
+import { useState } from 'react';
+import { addItem } from '../api/firebase';
+
 export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+	// sets initial default values in form fields and deconstruct form field variables
+	const initialState = { itemName: '', estimate: '7' };
```

Comment:
Great idea to define `initialState` as it's own `const`! Drastically improves readability 👍 

I'd suggest defining `initialState` outside of the `AddItem` component as shown below. Every time the `AddItem` component is rendered, it will be creating another `initialState` constant in memory. Since it's not changing, it's better to move this outside of the component to become a global variable within this file.

```javascript
const initialState = { itemName:  '', estimate: '7'};

export function AddItem() {
  ...
}
```

Reply:
Is there a reason why you decided to define the estimate as a `string` instead of a `number`?

Reply:
Wow, so I think I've been living by this unspoken rule that everything has to live inside of the component. Which apparently is not a rule at all! So does declaring it outside of the component improve performance/speed because it's one less task to perform with each render? Or am I misunderstanding the benefit?

As for the estimate, as I recall we tried to set the value to a number and we weren't able to for some reason. This was one of many stumbling blocks along the way that I think we just addressed with a pragmatic choice, not a strategic one, largely in the interest of time. Maybe it would have been better to ask for help at that point. Please correct me if you have a clearer or different recollection of why we did that, @a-cole55 .

@jendevelops I think both of these points speak to a bigger pattern in what I see myself doing, not just with TCL, but in general with my code: I can often make things work, but the ways in which I make them work may be less than optimal (putting it nicely) and the decisions behind those ways are often dictated by the limits of my current skills, not by a well-thought-out plan and design for the task I'm trying to accomplish. Maybe this is just part of the learning curve of being a new dev, but it would be nice to move away from "I did X because it was the only way I know how to do X" and toward "I did X because [fill in the blank with a strong rationale]." There isn't really a direct question here, but maybe you and the other mentors have thoughts or reflections from your own experience that speak to this. Thank you!

Reply:
> Wow, so I think I've been living by this unspoken rule that everything has to live inside of the component. Which apparently is not a rule at all! So does declaring it outside of the component improve performance/speed because it's one less task to perform with each render? Or am I misunderstanding the benefit?

Essentially under the hood every time you re-render, if `initialState` lives within the component, there will be a new `initialState` constant in memory for every re-render. In this case performance improvements/ speed improvements are minimal. But in other cases when the constant is large, or a computationally heavy value needs to be calculated to get the value you want, benefits will be much more obvious. Another thing to consider is that if you need a stable reference to something, a constant that is defined within the component probably won't fit the bill unless wrapped with `useMemo` or some other hook. But since this value is relatively small and constant, it's a better candidate to just move out of the component instead of using `useMemo`.

@tmflower It seems like [Firebase supports types other than `string`](https://firebase.google.com/docs/firestore/manage-data/data-types). What kind of errors were you expriencing?

Reply:
@jendevelops Thanks for clarifying!

The problem with numbers was in the form itself (radio button value property), not Firebase. @a-cole55  dug into this to better understand the problem and she is planning to share about it tomorrow.